designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentServiceImpl.scala

```diff
@@ -91,41 +105,58 @@ class DeploymentServiceImpl(
-          case Success(validationResult) =>
-            // we notify of deployment finish/fail only if initial validation succeeded
-            val deploymentFuture = runDeploymentActionWithNotifications(
-              actionType,
-              actionId,
-              processIdWithName,
-              versionOnWhichActionIsDone,
-              validatedComment,
-              buildInfoProcessIngType
-            ) {
-              dispatcher
-                .deploymentManagerUnsafe(processDetails.processingType)
-                .deploy(
-                  validationResult.processVersion,
-                  validationResult.deploymentData,
-                  validationResult.resolvedScenario,
-                  savepointPath
-                )
+    val actionType        = ProcessActionType.Deploy
+    val deploymentComment = request.params.get("comment")
+
+    isOfActionType(request.name, actionType) match {
+      case Some(_) =>
+        validateDeploymentComment(deploymentComment).flatMap { validatedComment =>
+          checkCanPerformActionAndAddInProgressAction[CanonicalProcess](
+            processIdWithName,
+            actionType,
+            d => Some(d.processVersionId),
+            d => Some(d.processingType)
+          ).flatMap { case (processDetails, actionId, versionOnWhichActionIsDone, buildInfoProcessIngType) =>
+            validateBeforeDeploy(processDetails, actionId).transformWith {
+              case Failure(ex) =>
+                dbioRunner.runInTransaction(actionRepository.removeAction(actionId)).transform(_ => Failure(ex))
+              case Success(validationResult) =>
+                // we notify of deployment finish/fail only if initial validation succeeded
+                val deploymentFuture = runDeploymentActionWithNotifications(
+                  actionType,
+                  actionId,
+                  processIdWithName,
+                  versionOnWhichActionIsDone,
+                  validatedComment,
+                  buildInfoProcessIngType
+                ) {
+                  dispatcher
+                    .deploymentManagerUnsafe(processDetails.processingType)
+                    .deploy(
+                      validationResult.processVersion,
+                      validationResult.deploymentData,
+                      validationResult.resolvedScenario,
+                      savepointPath
+                    )
+                }
+                Future.successful(deploymentFuture)
             }
-            Future.successful(deploymentFuture)
+          }
         }
-      }
+      case _ =>
+        Future.failed(
+          new IllegalArgumentException("Got a request for: " + request.name + " but ended up trying to deploy process")
+        )
+    }
+  }
+
+  private def isOfActionType(name: ScenarioActionName, actionType: ProcessActionType): Option[Unit] = {
+    if (name == ScenarioActionName(actionType)) {
+      Some(())
+    } else {
+      None
```

Comment:
I think this method could be removed and simple `if-else` would be sufficient and more readable.